### ISSUE TITLE
Suggested Integrations

### ISSUE THREAD

**Issue by mattkrick** [ParabolInc/action#2788]
When a user assigns a task to another user & clicks the publish icon, they should see a list of reasonable places to publish.

There are a few states to consider:
- an assignee has never had a task integrated with them before
- an assignee has received a task before, but now the team has created a new project & the assignee has never had a task for that project assigned to them
- an assignee has had tasks integrated with 3+ services, but some are stale & no longer used

We should limit the list to 3 items to keep it easy.
- The first item suggested should be the most recent integration that has been used on that assignee. If no integration has ever been used with the assignee, then we should get all the integrations that the team uses & filter by the ones that the assignee has auth for. 
- The second item should be the 2nd most recent integration that has been used for that assignee. if that doesn't exist, then grab the next in the list for the team.
- The third item should be the 3rd most recent integration at long as it's newer than 30 days. else, it should be provided by the team.

In summary,
we don't need to check user auth if
- the user has 3 integrations that have all been used within the last 30 days
- the user uses all of the team integrations

else, we check that the user has an auth token for the service provider.
- i don't think we need to verify with the provider that their auth token gets them access to that particular project/repo. that's because in a real-time environment, that can change between the time that we send the message & the time the client gets it, which means we'll have to plan for an error state, anyways. in reality, if 2 people on the same team use jira, it's probably a safe bet that they both have access to the same stuff

**Comment by jordanh** [comment]
Classic "@mattkrick thinking out loud" issue. Iceboxing for the historical record.
